### PR TITLE
fix: aura reallocation on low fee chains

### PR DESCRIPTION
**Pull request by Tritium-VLK**
Don't try to reallocate aura on pools that are total incentives under the aura min.

### CODE REVIEW

File: fee_allocator/accounting/distribution.py [hunk 1]
```diff
@@ -119,21 +118,43 @@ def re_distribute_incentives(
...
-                incentives[pool_id]['bal_incentives'] -= min(amount_per_pool, _data['bal_incentives'])
+        debt_repaid = 0
+        ## Find how many pools ever could be over min_aura_incentive
+        pools_over_aura_min = [pool_id for pool_id, _data in incentives.items() if
+                               _data['aura_incentives'] >= min_aura_incentive]
+        num_pools_over_min = len(pools_over_aura_min)
+        ## Figure out how much to shift per pool using an even split
+        if num_pools_over_min == 0:
+            print(
+                f"WARNING: {incentives[pool_id]['chain']}:{pool_id} has no pools over min_aura_incentive, but owes {debt_to_aura_market} to the aura market.  Debt will not be repaid.")
```
Comment (gosuto-inzasheru) [inline fee_allocator/accounting/distribution.py:134]: is it worth it to `raise` here instead? how big of an erroneous csv can this produce?

File: fee_allocator/accounting/distribution.py [hunk 2]
```diff
@@ -119,21 +118,43 @@ def re_distribute_incentives(
...
+            amount_per_pool = 0
+        else:
+            amount_per_pool = round(debt_to_aura_market / num_pools_over_min, 4)
+        ## Check if all ov our
```
Comment (gosuto-inzasheru) [inline fee_allocator/accounting/distribution.py:138]: ?

File: fee_allocator/accounting/distribution.py [hunk 3]
```diff
@@ -119,21 +118,43 @@ def re_distribute_incentives(
...
+            #   on total_incentives instead of aura incentives
+            #   if (incentives['aura_incentives'] + amount_per_pool) < min_aura_incentive:
+            #         num_pools_over_min -= 1
+            # Distribute the aura_debt to the pools that are over the min_aura_incentive
+            if incentives[pool_id]['total_incentives'] > 0:
+                # TODO:  Need to think about edge cases here and watch them.
+                incentives[pool_id]['aura_incentives'] += min(amount_per_pool, incentives[pool_id]['bal_incentives'])
+                incentives[pool_id]['bal_incentives'] -= min(amount_per_pool, incentives[pool_id]['bal_incentives'])
+                debt_repaid += min(amount_per_pool, incentives[pool_id]['bal_incentives'])
+            if debt_to_aura_market - debt_repaid > 0:
```
Comment (gosuto-inzasheru) [inline fee_allocator/accounting/distribution.py:150]:
```suggestion
            if debt_to_aura_market - debt_repaid >= 0:
```

File: fee_allocator/accounting/distribution.py [hunk 4]
```diff
@@ -119,21 +118,43 @@ def re_distribute_incentives(
...
+                print(
+                    f"{incentives[pool_id]['chain']}:{pool_id}  remaining debt to aura market: {debt_to_aura_market}, Debt repaid: {debt_repaid}, debt remaining: {debt_to_aura_market - debt_repaid}")
     return incentives
 
-def add_last_join_exit(incentives: Dict[str, Dict], chain: Chains, alertTimeStamp: Optional[int] = None) -> Dict[str, Dict]:
+
+def add_last_join_exit(incentives: Dict[str, Dict], chain: Chains, alertTimeStamp: Optional[int] = None) -> Dict[
+    str, Dict]:
```
Comment (gosuto-inzasheru) [inline fee_allocator/accounting/distribution.py:157]: this looks off

Comment (gosuto-inzasheru) [reply fee_allocator/accounting/distribution.py:157]: time for black ci linter in this repo hehe

File: fee_allocator/accounting/recon.py [hunk 5]
```diff
@@ -58,7 +58,7 @@ def recon_and_validate(fees: dict, fees_to_distribute: dict, timestamp_now: int,
     aura_vebal_share = round(aura_incentives / (aura_incentives + bal_incentives), 4)
     if target_aura_vebal_share:
         # check that aura_vebal_share is within 5% of target
-        assert abs(aura_vebal_share - target_aura_vebal_share) < 0.05, f"Reconciliation failed. Aura veBAL share is not within 5% of target. Aura veBAL share: {aura_vebal_share}, Target: {target_aura_vebal_share}"
+        assert abs(aura_vebal_share - target_aura_vebal_share) < 0.20, f"Reconciliation failed. Aura veBAL share is not within 5% of target. Aura veBAL share: {aura_vebal_share}, Target: {target_aura_vebal_share}"
```
Comment (gosuto-inzasheru) [inline fee_allocator/accounting/recon.py:61]: isnt 20% a bit much? or was this for debugging? if not, text needs to be adjusted

**Review by gosuto-inzasheru — COMMENTED** [review]
left some comments

diff on the gsheet looks great!! lets see how she holds up next run :)

File: fee_allocator/accounting/recon.py (hunk 5 above)
Comment (Tritium-VLK) [inline fee_allocator/accounting/recon.py:61]:
```suggestion
        assert abs(aura_vebal_share - target_aura_vebal_share) < 0.05, f"Reconciliation failed. Aura veBAL share is not within 5% of target. Aura veBAL share: {aura_vebal_share}, Target: {target_aura_vebal_share}"
```

Comment (Tritium-VLK) [reply fee_allocator/accounting/recon.py:61]: Was for debugging, but needs to be edited before merge. Thanks.

File: fee_allocator/accounting/distribution.py (hunk 4 above)
Comment (Tritium-VLK) [reply fee_allocator/accounting/distribution.py:157]: I'm still trying to figure out how to get it to work in my IDE.  I thought I was using black, but this is what it did.  Feel free to setup the linkter.  Don't think we should block this though/that's another pr.

File: fee_allocator/accounting/distribution.py (hunk 2 above)
Comment (Tritium-VLK) [inline fee_allocator/accounting/distribution.py:138]:
```suggestion
```

File: fee_allocator/accounting/distribution.py (hunk 1 above)
Comment (Tritium-VLK) [reply fee_allocator/accounting/distribution.py:134]: The final check in the recon (that we adjusted from 20 to 5) will make sure that we don't get too far away from where we should be.  If we do we have to refactor more.  Hoping to refactor this whole thing before that happens.

Comment (gosuto-inzasheru) [reply fee_allocator/accounting/distribution.py:134]: got it

**Review by gosuto-inzasheru — APPROVED** [review]

**Comment by gosuto-inzasheru** [comment]
or maybe solve rounding issue in this pr as well?

**Pull request merged**